include/model.php

```diff
@@ -60,6 +60,7 @@ public function __construct( &$options ) {
 		add_action( 'update_option_home', array( $this, 'clean_languages_cache' ) );
 
 		add_filter( 'get_terms_args', array( $this, 'get_terms_args' ) );
+		add_filter( 'get_terms_orderby', array( $this, 'filter_language_terms_orderby' ), 10, 3 );
```

Comment:
I suggest to add this filter only when we need it in `get_languages_list()` and `has_languages()` and to remove it once it has done its work, to avoid lots of calls to this filter.

Reply:
Done in [0665014](https://github.com/polylang/polylang/pull/1041/commits/0665014f42bd585c4fcf28197f42087da9900567).
I created a new method that factorizes the common code from `get_languages_list()` and `has_languages()`. It's `protected` but maybe it could be useful as `public` someday.